tests/pages/courses/id.test.js

```diff
@@ -0,0 +1,198 @@
+    fireEvent.click(registerButton) // Simulate a click on the registration button
+    expect(registerUserInCohortInFirestore).toHaveBeenCalledWith('RU5mLpQrZZWlmftNSB2w', 'user-123') // Ensure user registration is triggered
+  })
+
+  it('shows Calendar buttons after subscribing to the course', async () => {
+    renderCourse() // Render the course component
+
+    const registerButton = await screen.findByText('subscribeNow') // Find the subscription button
+    expect(registerButton).toBeInTheDocument() // Ensure the subscription button is present
+    fireEvent.click(registerButton) // Simulate a click on the registration button
+
+    await waitFor(async () => {
+      const registeredUser = {
+        ...mockUser,
+        cohorts: [
+          {
+            cohort_id: 'RU5mLpQrZZWlmftNSB2w', // Simulate the user being registered in a cohort
+            course_id: 'Rust_State_Machine',
+          },
+        ],
+      }
+
+      getUserFromFirestore.mockResolvedValue(registeredUser) // Mock updated user data retrieval
+      const calendar = await screen.findByText('addToCalendar') // Ensure "Add to Calendar" button is present
+      const addToGoogleCalendar = await screen.findByText('addToGoogleCalendar') // Ensure "Add to Google Calendar" button is present
+
+      expect(calendar).toBeInTheDocument() // Check that calendar button is visible
+      expect(addToGoogleCalendar).toBeInTheDocument() // Check that Google Calendar button is visible
+    })
+  })
+
+  it('shows Discord buttons after subscribing to the course', async () => {
+    renderCourse() // Render the course component
+
+    const registerButton = await screen.findByText('subscribeNow') // Find the subscription button
+    expect(registerButton).toBeInTheDocument() // Ensure the subscription button is present
+    fireEvent.click(registerButton) // Simulate a click on the registration button
+
+    await waitFor(async () => {
+      const registeredUser = {
+        ...mockUser,
+        cohorts: [
+          {
+            cohort_id: 'RU5mLpQrZZWlmftNSB2w', // Simulate the user being registered in a cohort
+            course_id: 'Rust_State_Machine',
+          },
+        ],
+      }
+
+      getUserFromFirestore.mockResolvedValue(registeredUser) // Mock updated user data retrieval
+      const discord = await screen.findByText('connectYourDiscord') // Ensure "Connect Your Discord" button is present
+
+      expect(discord).toBeInTheDocument() // Check that Discord button is visible
+    })
+  })
+
+  it('shows Wallet buttons after subscribing to the course', async () => {
+    renderCourse() // Render the course component
+
+    const registerButton = await screen.findByText('subscribeNow') // Find the subscription button
+    expect(registerButton).toBeInTheDocument() // Ensure the subscription button is present
+    fireEvent.click(registerButton) // Simulate a click on the registration button
+
+    await waitFor(async () => {
+      const registeredUser = {
```

Comment:
the rule of thumb for repeated code should be: "am I going to use this code again? Twice or more?" If you reuse the code just once there is no need to optimize but if you repeat 3 or more you should improve the logic because nothing prevents you of repeating 4 or even more.